modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -434,12 +442,17 @@ def broadcast_axis_partitions(
         # Since we are already splitting the DataFrame back up after an
         # operation, we will just use this time to compute the number of
         # partitions as best we can right now.
-        if keep_partitioning:
+        if num_splits == "keep":
             num_splits = len(left) if axis == 0 else len(left.T)
         elif lengths:
             num_splits = len(lengths)
-        else:
+        elif num_splits == "auto":
             num_splits = NPartitions.get()
+        else:
+            ErrorMessage.catch_bugs_and_request_email(
+                failure_condition=not isinstance(num_splits, int),
+                extra_log=f"Expected `num_splits` to be an integer, got: {type(num_splits)} | {num_splits=}",
```

Comment:
Why is this character needed at the end?
```suggestion
                extra_log=f"Expected `num_splits` to be an integer, got: {type(num_splits)} | {num_splits}",
```

Reply:
it's needed so it would print pretty: "Expected `num_splits` to be an integer, got: <class 'float'> | num_splits=24.0"
instead of: "Expected `num_splits` to be an integer, got: <class 'float'> | 24.0"